database-postgres/queries.js

```diff
@@ -0,0 +1,227 @@
+            INSERT INTO attributes (color_id, material, length, width, description_id)
+            VALUES (${attributesData}, ${descriptionIdToBeInserted})
+          `
+            )
+            .then((response) => {
+              console.log('Inserted into attributes!');
+              return knex.postgresDB
+                .raw(
+                  `
+              INSERT INTO directions (directions, description_id)
+              VALUES (${directionsData}, ${descriptionIdToBeInserted})
+            `
+                )
+                .then((response) => {
+                  return knex.postgresDB.raw(
+                    `
+                INSERT INTO details (additional_details, description_id)
+                VALUES (${detailsData}, ${descriptionIdToBeInserted})
+              `
+                  );
+                });
+            });
+        });
+    });
+}
+
+function updateSpecifiedTableRow(
+  idToSearch,
+  targetColumn,
+  newDataToBeInserted
+) {
+  const validQueries = {
+    description: 'descriptions',
+    title: 'descriptions',
+    sku: 'descriptions',
+    primary_brand: 'descriptions',
+    days_to_ship: 'descriptions',
+    color_id: 'attributes',
+    material: 'attributes',
+    length: 'attributes',
+    width: 'attributes',
+    directions: 'directions',
+    additional_details: 'directions',
+  };
+  // Validate that the table exists
+  if (!validQueries[targetColumn]) {
+    return Promise.reject(`${targetColumn} column does not exist`);
+  }
+  // validate that if the column is sku that the input is a number
+  // and 1,000,000 - 9,000.000
+  if (targetColumn === 'sku') {
+    if (isNaN(newDataToBeInserted) === true) {
+      return Promise.reject(
+        `Invalid data type trying to be inserted into ${targetColumn}`
+      );
+    }
+    if (newDataToBeInserted.length !== 7) {
+      return Promise.reject(
+        `Invalid data type trying to be inserted into ${targetColumn}`
+      );
+    }
+  }
+  // validate that the color falls within 1 - 967 (the acceptable colors)
+  if (targetColumn === 'color_id') {
+    if (isNaN(newDataToBeInserted) === true) {
+      return Promise.reject(
+        `Invalid data type trying to be inserted into ${targetColumn}`
+      );
+    }
+    if (Number(newDataToBeInserted) < 0 || Number(newDataToBeInserted) > 967) {
+      return Promise.reject(`Invalid range given for the ${targetColumn}`);
+    }
+  }
+
+  return knex
+    .postgresDB(validQueries[targetColumn])
+    .where({ description_id: idToSearch })
+    .update(targetColumn, newDataToBeInserted);
+}
+
+function getDataForSpecifiedId(idToSearch) {
+  return knex.postgresDB
+    .raw(
+      `SELECT
+    descriptions.title,
+    descriptions.primary_brand,
+    descriptions.sku,
+    descriptions.days_to_ship,
+    attributes.material,
+    attributes.length,
+    attributes.width,
+    directions.directions,
+    details.additional_details,
+    colors.*, descriptions.description
+    from descriptions
```

Comment:
Does this impact the query if `from` is lower case?

Reply:
No it doesn't. Using uppercase for SQL is just a convention, you can use lowercase but I should probably change FROM to uppercase just to stick to that convention good eye.